### DISCUSSION TITLE
Cowrie Honeypot

### DISCUSSION THREAD

**Discussion by MrPorkChopxx** [telekom-security/tpotce#1424] [Q&A]
Hello,

I'm currently on version 22.04.0 and even though my Cowrie honeypot gets lots of hits, I noticed that "Cowrie - Top URI Downloads" and 
"Cowrie - Top Downloads" widgets never gets populated with any data.  It always displays "No Results Found".  Any suggestions? Thank you.

**Comment by Ganesh-403** [comment]
@MrPorkChopxx 
If the rest of the **Cowrie dashboards are updating** but only **"Cowrie - Top URI Downloads"** and **"Cowrie - Top Downloads"** always show **"No Results Found"**, then the dashboard itself is probably working—the issue is more likely that **no download events are being generated**.

Those visualizations are populated only when Cowrie records successful file download events (for example, an attacker successfully downloads a file via `wget` or `curl`). Simply receiving SSH logins, password guesses, or shell commands will not populate these widgets. Cowrie is capable of logging downloaded files and storing them under `var/lib/cowrie/downloads/` when such events occur. :contentReference[oaicite:0]{index=0}

I'd recommend checking the following:

1. **Verify that download events exist in Kibana**
   In **Discover**, search for Cowrie download-related events (for example, events generated by `wget`/`curl`) and inspect whether any documents contain download information.

2. **Inspect the Cowrie JSON log**
   ```bash
   docker logs cowrie
   ```
   or check the JSON log directly (depending on your T-Pot version):

   ```bash
   cat /data/cowrie/log/cowrie.json
   ```

   Look for events corresponding to successful downloads.

3. **Check whether any files have actually been captured**

   Cowrie stores downloaded files locally. If the download directory is empty, then the dashboard has nothing to visualize. :contentReference[oaicite:1]{index=1}

4. **Confirm the dashboard matches your T-Pot version**

   You're running **22.04.0**, and from what I could find, I couldn't find any documented issue stating that these two widgets are broken in that release. If the index mappings or dashboard objects were modified manually, it's also worth verifying that the visualizations still reference the correct fields.

One question that would help narrow this down:

- When you inspect `cowrie.json`, do you actually see any download events (e.g. after an attacker runs `wget` or `curl`), or are you only seeing login attempts and shell activity?

If there are download events in the JSON log but the dashboard remains empty, then the next step would be to inspect the indexed documents in Elasticsearch to determine whether those fields are being parsed correctly by Logstash.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
